### PR TITLE
Core code editor/builder components

### PR DESCRIPTION
**Pull request by scottlepp**
**What this PR does / why we need it**:

Migrate components from [experimental repo](https://github.com/grafana/grafana-experimental) to core grafana.

These "experimental" components are used throughout grafana.  Also, now used within the new sql editors ( mysql, postgres, mssql).  They don't seem "experimental" anymore so migrating to core grafana repo.

Part of #51476

### CODE REVIEW

**Comment by sunker** [comment]
@scottlepp I'm working on migrating from grafana/ui CodeEditor to grafana/experimental SQLEditor in the Redshift plugin. Work i scoped out in [this](https://github.com/grafana/redshift-datasource/issues/108) issue. As part of this, I'm planning to make some improvements to the SQLEditor. For example, I want the editor to provide out of the box autocompletion support in the case no custom completion item provider was provided. 

Moving the SQLEditor to core grafana seems like the right thing to do, but I wonder if it can wait two months or so until I'm done?

File: public/app/features/plugins/sql/standardSql/macros.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+/*
+  {
+    id: "$__timeGroupAlias(dateColumn, '5m', fill)",
+    name: "$__timeGroupAlias(dateColumn, '5m', fill)",
+    text: "$__timeGroup",
+    args: [COLUMN, RELATIVE_TIME_STRING , FILL_TYPE],
+    type: MacroType.Group,
+    description:
+      "Will be replaced by an expression usable in GROUP BY clause. For example, *cast(cast(UNIX_TIMESTAMP(dateColumn)/(300) as signed)*300 as signed),* with an added column alias",
```
Comment (dprokop) [inline public/app/features/plugins/sql/standardSql/macros.ts:77]: Let's remove this commented out code.

File: public/app/features/plugins/sql/standardSql/statementPositionResolversRegistry.ts [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        return false;
+      },
+    },
+
+    {
+      id: StatementPosition.FromKeyword,
+      name: StatementPosition.FromKeyword,
+      resolve: (currentToken, previousKeyword, previousNonWhiteSpace, previousIsSlash) => {
+        // cloudwatch specific commented out
+        // Boolean(previousKeyword?.value === SELECT && previousNonWhiteSpace?.isParenthesis()),
```
Comment (dprokop) [inline public/app/features/plugins/sql/standardSql/statementPositionResolversRegistry.ts:96]:
```suggestion
```

File: public/app/features/plugins/sql/standardSql/statementPositionResolversRegistry.ts [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    {
+      id: StatementPosition.AfterFrom,
+      name: StatementPosition.AfterFrom,
+      resolve: (currentToken, previousKeyword, previousNonWhiteSpace, previousIsSlash) =>
+        Boolean(
+          (previousKeyword?.value.toLowerCase() === FROM && previousNonWhiteSpace?.isDoubleQuotedString()) ||
+            (previousKeyword?.value.toLowerCase() === FROM && previousNonWhiteSpace?.isIdentifier()) ||
+            (previousKeyword?.value.toLowerCase() === FROM && previousNonWhiteSpace?.isVariable())
+          //  cloudwatch specific
+          // (previousKeyword?.value === SCHEMA && previousNonWhiteSpace?.is(TokenType.Parenthesis, ')'))
```
Comment (dprokop) [inline public/app/features/plugins/sql/standardSql/statementPositionResolversRegistry.ts:119]:
```suggestion
```

File: public/app/features/plugins/sql/standardSql/getStatementPosition.test.ts [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  //   assertPosition(query, position, StatementPosition.SchemaFuncFirstArgument);
+  // });
+
+  // test.each([
+  //   [singleLineFullQuery.query, { line: 1, column: 50 }],
+  //   [multiLineFullQuery.query, { line: 2, column: 23 }],
+  //   [singleLineTwoQueries.query, { line: 1, column: 50 }],
+  // ])('should be label key args within the schema func', (query: string, position: monacoTypes.IPosition) => {
+  //   assertPosition(query, position, StatementPosition.SchemaFuncExtraArgument);
+  // });
```
Comment (dprokop) [inline public/app/features/plugins/sql/standardSql/getStatementPosition.test.ts:200]:
```suggestion
```

**Review by dprokop — APPROVED** [review]
@sunker do you mind having a look at the code of the raw editor and identiying pieces of code that are used in the cloudwatch only? Like i.e. `getNamespaceToken`. I think I removed a lot when implementing this in experimental, but there still may be some leftovers.

@scottlepp - LGTM!

**Comment by sunker** [comment]
So this PR is moving everything from grafana/experimental to grafana core? Is that something all the plugin teams have agreed on? Has it been communicated to other teams? What will happen to grafana/experimental now? Will you deprecate it or is it just going to diverge from the components in grafana ui?

**Comment by torkelo** [comment]
@scottlepp @sunker @dprokop  

I think this could have needed some more discussion, and work before being added to @grafana/ui  , none of these components have storybook stories. They are in folder named "QueryEditor" but exported from root namespace (so this fact that they are for query editors is completely lost), Stack which is a general layout component should not be inside this QueryEditor folder. 

Stack is a really useful component and has nothing to do with QueryEditor and should be in the root components folder or in the Layout folder. I tried to add this before but with improvements of forwardref classname, storybook story etc: https://github.com/grafana/grafana/pull/45972 but got no reviews so closed it.

**Comment by dprokop** [comment]
@torkelo 😬 yes, definitely, @scottlepp @sunker and myself we had a lot of discussion about this being merged during summer (unfortunately after the fact). But I thought we agreed to revert this?

**Comment by torkelo** [comment]
@scottlepp @sunker @dprokop @mckn 

This is going out in 9.2 but there is still time to revert this but it's going to be very painful, will probably need a manual revert in main and a separate manual revert in 9.2.x (doing a normal git revert on this will probably lead to many conflicts, same with backport)

**Comment by scottlepp** [comment]
Do we really need to revert it?  Sounds like we just need to move them to a better location in grafana?

I was seeing a problem.  We were using Stack and other components from "experimental" throughout grafana.  Experimental doesn't have storybook either.  At what point are "experimental" components no longer experimental?

Does not having "experimental" components used throughout grafana make grafana itself experimental?  We can just keep adding components to experimental, without storybook, and use them in grafana wherever we want?

**Comment by torkelo** [comment]
@scottlepp well the big difference is that we can change components in grafana/experimental without breaking plugins, even when we are using them in many places in core and in plugins.

Adding a component to grafana/ui makes it much more public and available to all plugin developers (why storybook and component docs are needed) and we lose the ability to make any breaking changes.

Think promoting a component from grafana/experimental to grafana/ui needs some more quality tests other than it's used widely used in core.

**Comment by torkelo** [comment]
@scottlepp if you think we can quickly solve the naming spacing issue (renaming all components to make it clear these are for QueryEditors) quickly without revert then maybe that is an option. 

tack component could need a better name (FlexLayout or something)

**Comment by torkelo** [comment]
@scottlepp but I agree this was the right step to take for some of these components. But was just a done a bit too quick, without a discussion on component naming, component refinement, component location within grafana/ui, and some basic storybook stories.

**Comment by sunker** [comment]
Yeah I also think this happened a bit to fast. Also, some of the types that were moved from grafana/experimental to core grafana are not exported by grafana/data, so now where in a situation where external plugins need to pin to an old version of grafana/experiemental to be able to use the new query editor UI.

**Pull request merged**